FoundationsMasteryTests/CustomStringTests.cs

```diff
@@ -9,38 +11,101 @@ public class CustomStringTests
         [TestMethod]
         public void EnsureICanCreateAnInstance()
         {
+            //Arrange
+            char[] mychars = new char[] { 'a', 'b', 'c' };
+            CustomString myString = new CustomString(mychars);
+
+            //Act (nothing to Act for this)
+
+            //Assert
+            Assert.IsNotNull(myString);
         }
 
         [TestMethod]
         public void EnsureICanPassInACharEnumerable()
         {
+            //Arrange
             char[] mychars = new char[] { 'a', 'b', 'c' };
             CustomString myString = new CustomString(mychars);
 
-            //int expected_length = ??
-            //int actual_length = ??
+            //Act
+            int expected_length = 3;
+            int expected_length_other = mychars.Length;
+            int actual_length = myString.Length;
 
+
+            //Assert
             Assert.AreEqual(expected_length, actual_length);
+            Assert.AreEqual(expected_length_other, actual_length);
         }
 
         [TestMethod]
         public void EnsureIHaveContents()
         {
+            // 1. Assert that the Contents property is the correct 'type'
+            // 2. Assert that the returned Contents is the coorect length
+            /* #2 is redudant to tests above */
+
+            //Arrange
             char[] mychars = new char[] { 'a', 'b', 'c' };
             CustomString myString = new CustomString(mychars);
 
-            // 1. Assert that the Contents property is the correct 'type'
-            // 2. Assert that the returned Contents is the coorect length
+            //Act
+            int expected_length = mychars.Length;
+            int actual_length = myString.Length;
+
+            //Assert
+            Assert.AreEqual(expected_length, actual_length);
+            Assert.IsInstanceOfType(mychars, typeof(Char[]));            
```

Comment:
We know `mychars` here must be of type `char[]` - we're interested in the type of `myString.Contents`